skimage/metrics/overlap.py

```diff
@@ -0,0 +1,251 @@
+"""
+Module defining a number of functions to quantify the overlap between shapes.
+
+For instance BoundingBoxes used for object-detection.
+"""
+from __future__ import annotations
+
+import numpy as np
+
+
+class BoundingBox:
+    """
+    Construct an axis-aligned Bounding-Box
+    consisting of top left and bottom right corners.
+
+    The contructor uses the (r,c,..) coordinates for the top left corner and
+    either the coordinates of the botton right corner or the BoundingBox
+    dimensions (height, width,...).
+
+    BoundingBoxes can have 2 (rectangle), 3 (3D BoundingBox) or more dimensions.
+
+    Parameters
+    ----------
+    top_left : array-like of ints or floats
+        (r,c)-coordinates for the top left corner of the BoundingBox.
+
+    bottom_right : array-like of ints or floats, optional
+        (r,c)-coordinates for the bottom right corner of the BoundingBox.
+
+    dimensions : array-like of ints or floats, optional
+        dimensions of the BoundingBox (height, width). The default is None.
+
+    Raises
+    ------
+    ValueError
+        If neither or both of bottom_right and dimensions are provided.
+
+    Attributes
+    ----------
+    top_left : array of int or float
+        The top left corner of the BoundingBox.
+    bottom_right : array of int or float
+        The bottom right corner of the BoundingBox.
```

Comment:
Missing dimensions and a bunch of others

Reply:
the latest commit documents the other attributes/properties